torchtune/_cli/run.py

```diff
@@ -0,0 +1,165 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import argparse
+import runpy
+import sys
+import textwrap
+
+from pathlib import Path
+from typing import Optional
+
+import torchtune
+
+from torch.distributed.run import get_args_parser as get_torchrun_args_parser, run
+from torchtune import Config, get_all_recipes, Recipe
+from torchtune._cli.subcommand import Subcommand
+
+ROOT = Path(torchtune.__file__).parent.parent
+
+
+class Run(Subcommand):
```

Comment:
I'm staring at this and still not sure what tune run is getting us? Better UX than torchrun?

BTW can we add some examples of how I'd run multiple finetune jobs on the same host on different cuda devices? I know this needs some wrangling of the underlying torchrun params. But would be a huge QoL improvement if we can just call this out in the examples.

Reply:
It's a more consistent UX (everything runs with subcommands of argparse) and in general, this PR adds more testing for running of recipes.

On the second point sure, will reach out to you directly on this.

Reply:
```
tune run --nproc_per_node 2 --rdzv-backend=c10d --rdzv-endpoint:localhost:20000 full_finetune_distributed --config llama2/7B_full

# Then you can launch
tune run --nproc_per_node 2 --rdzv-backend=c10d --rdzv-endpoint:localhost:<OTHER-PORT> full_finetune_distributed --config llama2/7B_full
```